src/imageops/fast_blur.rs

```diff
@@ -0,0 +1,144 @@
+use std::cmp::{max, min};
+
+use crate::{ImageBuffer, Pixel, Primitive};
+
+/// Approximation of Gaussian blur after
+/// Kovesi, P.:  Fast Almost-Gaussian Filtering The Australian Pattern
+/// Recognition Society Conference: DICTA 2010. December 2010. Sydney.
+pub fn fast_blur<P: Pixel>(
+    image_buffer: &ImageBuffer<P, Vec<P::Subpixel>>,
+    sigma: f32,
+) -> ImageBuffer<P, Vec<P::Subpixel>> {
+    let (width, height) = image_buffer.dimensions();
+    let mut samples = image_buffer.as_flat_samples().samples.to_vec();
+    let num_passes = 3;
+
+    let boxes = boxes_for_gauss(sigma, num_passes);
+
+    for radius in boxes.iter().take(num_passes) {
+        let horizontally_blurred_transposed = my_fast_horizontal_blur::<P::Subpixel>(
+            &samples,
+            width as usize,
+            height as usize,
+            *radius,
+            P::CHANNEL_COUNT as usize,
+        );
+        samples = my_fast_horizontal_blur::<P::Subpixel>(
+            &horizontally_blurred_transposed,
+            height as usize,
+            width as usize,
+            *radius,
+            P::CHANNEL_COUNT as usize,
+        );
+    }
+    ImageBuffer::from_raw(width, height, samples).unwrap()
+}
```

Comment:
`width = 0` as well as `height = 0` are 'normal' cases for input data. I think this misbehaves in a few of these cases in differing ways. (Overflow in `x - 1` in `extended_f` for instance). A test for that case would be good.

Reply:
See below.